src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java

```diff
@@ -47,7 +49,8 @@ public ExtendedAdDTO getInfoByAd(int id) {
     }
 
     @Override
-    public void deleteUser(int id) {
+    public void deleteAd(int id) {
+        commentRepository.deleteAllByAdEntity(adRepository.findById(id).get());
```

Comment:
```suggestion
        commentRepository.deleteAllByAdEntity(adRepository.findById(id).orElseThrow(...));
```
get без проверки - будет исключение, если не будет объявления